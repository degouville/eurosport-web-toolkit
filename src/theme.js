import { transparentize, rgba } from 'polished';
import {
  plausibleViolet,
  crashingAmber,
  coreLightMinus1,
  significantAzure,
  featureOneMinus2,
  actionTwoDarkBase,
  coreDarkPlus1,
  coreNeutral2,
  coreNeutral4,
  coreNeutral8,
  coreLightBase,
  dodgerBlue2,
  actionTwoLightBase,
  melanzane,
  coreDarkBase,
  hydroelectricTar,
  coreNeutral6,
  coreNeutral1,
  actionOneDarkMinus1,
  actionOneDarkBase,
  ebonyClay,
  martinique,
  suitGray,
  blackPearl,
} from './colors';
import { fontAlphaHeadline, fontInterUi } from './typography';

export default {
  underlineLink: {
    fontFamily: fontInterUi,
    color: coreNeutral4,
    hoverColor: coreNeutral2,
  },
  link: {
    color: featureOneMinus2,
  },
  button: {
    primary: {
      color: coreLightMinus1,
      backgroundColor: actionTwoDarkBase,
    },
    secondary: {
      color: featureOneMinus2,
    },
    form: {
      color: coreLightMinus1,
      backgroundColor: dodgerBlue2,
    },
  },
  input: {
    backgroundColor: coreDarkPlus1,
    borderFocused: coreNeutral4,
    border: coreNeutral8,
    placeholder: {
      color: coreNeutral4,
      fontFamily: fontInterUi,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  login: {
    separator: {
      primary: coreLightBase,
      secondary: coreNeutral8,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
    title: {
      color: coreLightMinus1,
      fontFamily: fontAlphaHeadline,
    },
  },
  banner: {
    error: {
      borderColor: actionTwoLightBase,
      backgroundColor: melanzane,
      textColor: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  tab: {
    active: {
      color: coreLightMinus1,
      borderColor: featureOneMinus2,
    },
    borderColor: transparentize(0.5, coreLightMinus1),
  },
  typo: {
    primary: {
      color: coreLightMinus1,
    },
  },
  playerControls: {
    dropdown: {
      background: transparentize(0.1, coreDarkBase),
      border: transparentize(0.2, coreNeutral6),
    },
    seek: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: false,
    },
    volume: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: true,
    },
    bar: {
      backgroundColor: transparentize(0.4, coreDarkBase),
    },
    separator: transparentize(0.5, coreNeutral4),
    liveIcon: {
      fontFamily: fontAlphaHeadline,
      backgroundColor: actionTwoDarkBase,
      color: coreLightBase,
    },
    text: {
      fontFamily: fontAlphaHeadline,
      color: coreLightBase,
    },
  },
  dropdown: {
    label: {
      color: coreLightMinus1,
    },
    list: {
      default: {
        color: coreNeutral2,
      },
      selected: {
        color: coreLightMinus1,
      },
    },
    checkMark: {
      color: significantAzure,
    },
    background: hydroelectricTar,
    border: transparentize(0.2, coreNeutral6),
  },
  scrollbar: {
    color: plausibleViolet,
    background: crashingAmber,
  },
  statistics: {
    comparison: {
      colorOne: actionOneDarkMinus1,
      colorTwo: actionOneDarkBase,
      background: rgba(coreLightMinus1, 0.1),
    },
    background: {
      one: ebonyClay,
      two: martinique,
    },
    fringe: rgba(coreLightMinus1, 0.05),
  },
  marketingChecklist: {
    title: {
      color: coreLightMinus1,
    },
    subtitle: {
      color: suitGray,
    },
    check: {
      color: coreLightMinus1,
    },
  },
  videoPlayerModal: {
    closeIcon: {
      fill: coreLightMinus1,
    },
  },
  loginWithMarketing: {
    mobileBackground: blackPearl,
  },
};
