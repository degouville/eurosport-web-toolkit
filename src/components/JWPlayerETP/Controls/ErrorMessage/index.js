import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';
import * as breakpoints from 'src/breakpoints';
import ErrorIcon from 'src/assets/error-icon.component.svg';
import { HorizontalSeparator } from 'src/components/JWPlayerETP/Controls/UI/separators';

const ErrorMessage = ({ message }) => (
  <MainContainer>
    <MessageContainer>
      <Icon />
      <Text>{message}</Text>
    </MessageContainer>
    <HorizontalSeparator />
  </MainContainer>
);

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

const MainContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  background-color: ${({ theme }) => theme.playerControls.errorMessage.backgroundColor};
`;

const MessageContainer = styled.div`
  width: 70%;
  display: flex;
  align-items: center;
  flex-direction: column;
`;

const Icon = styled(ErrorIcon)`
  path {
    fill: ${({ theme }) => theme.playerControls.errorMessage.iconColor};
  }
  width: 72px;
  ${breakpoints.medium(css`
    width: 96px;
  `)}
`;

const Text = styled.p`
  margin: 19px 0 19px;
  color: ${({ theme }) => theme.playerControls.errorMessage.textColor};
  font-family: ${({ theme }) => theme.banner.error.fontFamily};
  font-size: 18px;
  line-height: 27px;
  text-align: center;
  ${breakpoints.medium(css`
    margin: 27px 0 27px;
    font-size: 24px;
    line-height: 36px;
  `)}
`;

export default ErrorMessage;
