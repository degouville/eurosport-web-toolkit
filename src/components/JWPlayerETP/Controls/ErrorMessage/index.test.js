import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'emotion-theming';
import theme from 'src/theme';
import ErrorMessage from './index';

describe('ErrorMessage', () => {
  it('Should match snapshot', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ErrorMessage message="Error" />
      </ThemeProvider>
    );

    expect(wrapper.find(ErrorMessage)).toMatchSnapshot();
  });
  it('Should display correct error message', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ErrorMessage message="Error Message" />
      </ThemeProvider>
    ).find(ErrorMessage);
    expect(wrapper.text()).toEqual('Error Message');
  });
});
