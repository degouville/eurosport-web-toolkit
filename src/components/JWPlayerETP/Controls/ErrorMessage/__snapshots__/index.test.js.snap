// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Should match snapshot 1`] = `
.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  background-color: #151726;
}

.emotion-6 {
  width: 70%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  width: 72px;
}

.emotion-0 path {
  fill: #2D304A;
}

@media (min-width:700px) {
  .emotion-0 {
    width: 96px;
  }
}

.emotion-4 {
  margin: 19px 0 19px;
  color: #F8F8F9;
  font-family: font-family:'Inter UI',-apple-system,-system-ui,system-ui,BlinkMacSystemFont,'Segoe UI',Helvetica,Arial,sans-serif;
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
  font-smooth: auto;
  font-size: 18px;
  line-height: 27px;
  text-align: center;
}

@media (min-width:700px) {
  .emotion-4 {
    margin: 27px 0 27px;
    font-size: 24px;
    line-height: 36px;
  }
}

.emotion-8 {
  height: 1px;
  width: 100%;
  background-color: rgba(151,152,166,0.5);
}

<ErrorMessage
  message="Error"
>
  <Styled(div)>
    <div
      className="emotion-10 emotion-11"
    >
      <Styled(div)>
        <div
          className="emotion-6 emotion-7"
        >
          <Styled(ErrorIcon)
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <ErrorIcon
              className="emotion-0 emotion-1"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <svg
                className="emotion-0 emotion-1"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink"
              >
                <defs>
                  <path
                    d="M16 26c5.523 0 10-4.477 10-10S21.523 6 16 6 6 10.477 6 16s4.477 10 10 10zm0 2C9.373 28 4 22.627 4 16S9.373 4 16 4s12 5.373 12 12-5.373 12-12 12zm-.027-8.73c.41 0 .753.131 1.026.393.274.261.41.588.41.98s-.136.719-.41.98-.616.393-1.026.393-.753-.131-1.027-.393c-.273-.261-.41-.588-.41-.98s.137-.719.41-.98.616-.393 1.027-.393zm-.565-10.392h1.13c.205 0 .367.058.487.172.12.114.162.253.128.417l-.36 8.235c0 .163-.059.302-.179.416a.608.608 0 0 1-.436.172h-.41a.608.608 0 0 1-.437-.172c-.12-.114-.18-.236-.18-.367l-.359-8.284a.422.422 0 0 1 .129-.417c.12-.114.282-.172.487-.172z"
                    id="a"
                  />
                </defs>
                <g
                  fill="none"
                  fillRule="evenodd"
                  transform="translate(-4 -4)"
                >
                  <mask
                    fill="#fff"
                    id="b"
                  >
                    <use
                      xlinkHref="#a"
                    />
                  </mask>
                  <use
                    fill="#979797"
                    fillRule="nonzero"
                    xlinkHref="#a"
                  />
                  <g
                    fill="#FF0043"
                    mask="url(#b)"
                  >
                    <path
                      d="M0 0h32v32H0z"
                    />
                  </g>
                </g>
              </svg>
            </ErrorIcon>
          </Styled(ErrorIcon)>
          <Styled(p)>
            <p
              className="emotion-4 emotion-5"
            >
              Error
            </p>
          </Styled(p)>
        </div>
      </Styled(div)>
      <Styled(div)>
        <div
          className="emotion-8 emotion-9"
        />
      </Styled(div)>
    </div>
  </Styled(div)>
</ErrorMessage>
`;
